test/withStyles_test.js

```diff
@@ -0,0 +1,52 @@
+import { expect } from 'chai';
+import sinon from 'sinon-sandbox';
+
+import * as detectHooks from '../src/utils/detectHooks';
```

Comment:
using `import * as` and sinon does work, and need not be a blocker, but it's an abuse of babel's interop, and will stop working once node has native ESM.

Reply:
@ljharb How can I mock this with mocha? 🤔 I take a lot of the testing infra for granted we have at Airbnb and don't really know how to do it. 😅 Any thoughts?

Reply:
That's kind of the trick; module mocking requires something as intrusive as jest, or some pretty messy hacks. If `detectHooks` is a function, then you could stub properties on `React` itself, and directly fake the things it checks for.